src/pages/VenuePage/VenuePage.tsx

```diff
@@ -317,68 +310,7 @@ const VenuePage = () => {
     history.push(`/account/profile?venueId=${venueId}`);
   }
 
-  let template;
-  let fullscreen = false;
-  switch (venue.template) {
-    case VenueTemplate.jazzbar:
-      template = <JazzbarRouter />;
-      break;
-    case VenueTemplate.friendship:
-      template = <FriendShipPage />;
-      break;
-    case VenueTemplate.partymap:
-      template = <PartyMapRouter />;
-      break;
-    case VenueTemplate.artpiece:
-      template = <ArtPiece />;
-      break;
-    case VenueTemplate.themecamp:
-      template = <CampRouter />;
-      break;
-    case VenueTemplate.playa:
-    case VenueTemplate.preplaya:
-      template = <PlayaRouter />;
-      fullscreen = true;
-      break;
-    case VenueTemplate.zoomroom:
-    case VenueTemplate.performancevenue:
-    case VenueTemplate.artcar:
-      if (venue.zoomUrl) {
-        window.location.replace(venue.zoomUrl);
-      }
-      template = (
-        <p>
-          Venue {venue.name} should redirect to a URL, but none was set.
-          <br />
-          <button
-            role="link"
-            className="btn btn-primary"
-            onClick={() => history.goBack()}
-          >
-            Go Back
-          </button>
-        </p>
-      );
-      break;
-    case VenueTemplate.audience:
-      template = <AudienceRouter />;
-      fullscreen = true;
-      break;
-    case VenueTemplate.avatargrid:
-      template = <AvatarRouter />;
-      break;
-    case VenueTemplate.conversationspace:
-      template = <ConversationSpace />;
-      break;
-
-    case VenueTemplate.firebarrel:
-      template = <FireBarrel />;
-      break;
-  }
-
-  return (
-    <WithNavigationBar fullscreen={fullscreen}>{template}</WithNavigationBar>
-  );
+  return <TemplateWrapper venue={venue} />;
```

Comment:
What do we gain by extracting this into it's own component here by the way? It seems unrelated to the specified purposes of this PR?

Reply:
Agreed, unrelated to the PR, but that was my urge to decouple that. Decoupling components helps in reducing the cognitive efforts needed to understand what is going on in this component.

Reply:
https://github.com/sparkletown/sparkle/pull/1042#discussion_r551769022

The problem with that is that `fullscreen` value is calculated together with `venue` value. So, it would require additional effort to decouple that

Reply:
I do understand the desire to de-couple/simplify things, but in this case I feel like it kind of just adds a layer of abstraction without really 'gaining much'. It's also unrelated to the stated topic of the PR, so it's kind of out of scope (OOS).

I feel like the solution as implemented isn't really any better than the old way, and in a way might even make it worse in a sense.. I won't block the PR on it, and I guess we don't need to revert it.. but yeah, probably keep these sorts of things to their own PR in future so that we can assess it in isolation, and so any discussion/refinement doesn't block the more important changes being made here.

Reply:
It is not finished in any way. I just started doing that and then realised that it would take too much time and just aborted the plan. Didn't revert it back, because I think it is a good start to move that into another file and to work on that isolated part to make it better. Agreed with what was said above.

Reply:
_nods_ fair enough, makes sense.